plugins/askrene/askrene.c

```diff
@@ -727,25 +712,21 @@ static struct command_result *json_askrene_unreserve(struct command *cmd,
 {
 	struct reserve_path *path;
 	struct json_stream *response;
-	size_t num;
 	struct askrene *askrene = get_askrene(cmd->plugin);
 
 	if (!param(cmd, buffer, params,
 		   p_req("path", param_reserve_path, &path),
 		   NULL))
 		return command_param_failed();
 
-	num = reserves_remove(askrene->reserved, path->scidds, path->amounts,
-			      tal_count(path->scidds));
-	if (num != tal_count(path->scidds)) {
-		const struct reserve *r = find_reserve(askrene->reserved, &path->scidds[num]);
-		return command_fail(cmd, JSONRPC2_INVALID_PARAMS,
-				    "Underflow unreserving %zu: %s amount %s (%zu reserved, amount %s)",
-				    num,
-				    fmt_short_channel_id_dir(tmpctx, &path->scidds[num]),
-				    fmt_amount_msat(tmpctx, path->amounts[num]),
-				    r ? r->num_htlcs : 0,
-				    r ? fmt_amount_msat(tmpctx, r->amount) : "none");
+	for (size_t i = 0; i < tal_count(path); i++) {
```

Comment:
Also here `path` is a pointer to a single path. The loop should go up to the size of the path `tal_count(path->scidds)`.

Reply:
Yes, not sure what I was thinking there!  Removed...